src/libsyntax/parse/parser.rs

```diff
@@ -4511,7 +4511,19 @@ impl<'a> Parser<'a> {
                     if self.token == token::Semi {
                         unused_attrs(&attrs, self);
                         self.bump();
-                        return Ok(None);
+                        let mut last_semi = lo;
+                        while self.token == token::Semi {
+                            last_semi = self.token.span;
+                            self.bump();
+                        }
+                        return Ok(Some(Stmt {
+                            id: ast::DUMMY_NODE_ID,
+                            span: lo.to(last_semi),
+                            node: StmtKind::Semi(self.mk_expr(lo.to(last_semi),
+                                ExprKind::Tup(Vec::new()),
```

Comment:
This feels a little hacky, but at the same time, it seems relatively harmless. Can you add a comment pointing out that we're encoding a string of `;`s as a `()`?

Reply:
Yeah it does feel a bit hacky, not exactly ideal. Any ideas for a cleaner solution?

That aside, I added a comment in the latest push

Reply:
It feels messier to add a new `ExprKind` just for this case, so I think this encoding is reasonable.